api.go

```
@@ -1164,3 +1235,151 @@ func (c *Client) GetCreds() (credentials.Value, error) {
+			}
+			if hasQuery("acl") {
+				return "GetObjectAcl"
+			}
+			if hasQuery("legal-hold") {
+				return "GetObjectLegalHold"
+			}
+			if hasQuery("retention") {
+				return "GetObjectRetention"
+			}
+			return "GetObject"
+		} else {
```

golangci-lint (2.12.2)

[warning] indent-error-flow: if block ends with a return statement, so drop this else and outdent its block

(revive)